### ISSUE TITLE
[BOOM-5101] Voucher free shipping show 0.00% instead of free sipping

### ISSUE THREAD

**Issue by prestonBot** [PrestaShop/PrestaShop#9702]
## SPECS

Cart rule with free shipping only: Display free shipping instead of 0,00%
Cart rule with free shipping + discount (amount or %): Display price of the carrier + Discount amount added
Cart rule with % discount: Display the discount amount            

![image](https://user-images.githubusercontent.com/13449658/67876296-f5f8b100-fb37-11e9-81b2-076c75f642c0.png)

You should always see the same value between the "discount(s)" line and the "cart rule" line if there is only one cart rule in the cart

----------------------
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-5101).

- _**Reporter:**_ PN
- _**Created at:**_ Thu, 15 Mar 2018 09:27:45 +0100

<p>Hello,</p>

<p> </p>

<p><b>Type of view discounts in cart when is discount action set as free shipping. Why is showed: -0.00%? Better will be show free shipping instead of -0.00%.</b></p>

<p>Thank you</p>

<p>Best regards</p>

<p>PN</p>
        
* How to reproduce the issue ?

Catalog, Discounts, Cart Rules
<br/>

<br/>
for:
<br/>
1) Set Highlight: YES
<br/>
2) Leave field code blank
<br/>
3) Set actions: Free shipping: YES, none

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5101#comment-164244).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Thu, 15 Mar 2018 10:42:20 +0100

<p>Salut,</p>

<p>Que pensez-vous de sa remarque ?<br/>
J'ai pu reproduire le problème avec 1.7.3.0<br/>
 <span class="image-wrap" style=""><a id="66034_thumb" href="http://forge.prestashop.com/secure/attachment/66034/66034_1730.png" title="1730.png" file-preview-type="image" file-preview-id="66034" file-preview-title="1730.png"><img src="http://forge.prestashop.com/secure/thumbnail/66034/_thumb_66034.png" style="border: 0px solid black" /></a></span> </p>

<p>Merci!</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5101#comment-164253).

- _**Author:**_ marion_francois
- _**Created at:**_ Thu, 15 Mar 2018 11:55:40 +0100

<p>Hello <a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=PN" class="user-hover" rel="PN">PN</a></p>

<p>About voucher without code not automatically applied in cart, this is the fix: <a href="https://github.com/PrestaShop/PrestaShop/pull/8411" class="external-link" rel="nofollow">https://github.com/PrestaShop/PrestaShop/pull/8411</a><br/>
 You can follow this ticket for more information: <a href="http://forge.prestashop.com/browse/BOOM-3307" class="external-link" rel="nofollow">http://forge.prestashop.com/browse/BOOM-3307</a></p>

<p>About voucher highlighting problem, please follow this ticket: <a href="http://forge.prestashop.com/browse/BOOM-4544" class="external-link" rel="nofollow">http://forge.prestashop.com/browse/BOOM-4544</a></p>

<p>I add the -0.00% to change for "free shipping" in our debug roadmap</p>

<p>Regards</p>

<p> </p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5101#comment-166562).

- _**Author:**_ hubbobubbo
- _**Created at:**_ Sun, 22 Apr 2018 21:24:14 +0200

<p>I am cheering for this fix as well. I am testing 1.7.3.1 and when a voucher with Free Shipping is applied it prints -0.00% instead of free shipping.</p>

<p>This is a regression from previous PS versions so it would be nice to have it working the expect way.</p>

<p>Any ideas on when this might happen?</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5101#comment-170913).

- _**Author:**_ pentarou
- _**Created at:**_ Thu, 5 Jul 2018 10:58:55 +0200

<p>I hope the problem will be resolved quickly, most customers will certainly not understand this and will therefore doubt whether they want to complete their order.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5101#comment-173278).

- _**Author:**_ marion_francois
- _**Created at:**_ Thu, 2 Aug 2018 09:30:16 +0200

<p>Fixed in 1.7.4.0 by: </p>

<p><a href="https://github.com/PrestaShop/PrestaShop/pull/8411" class="external-link" rel="nofollow">https://github.com/PrestaShop/PrestaShop/pull/8411</a> and <a href="https://github.com/PrestaShop/PrestaShop/pull/9055" class="external-link" rel="nofollow">https://github.com/PrestaShop/PrestaShop/pull/9055</a></p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-5101#comment-173312).

- _**Author:**_ pn
- _**Created at:**_ Thu, 2 Aug 2018 12:04:42 +0200

<p>Hello Marion,</p>

<p>point 3 is not corrected. (PS versios 1.7.4.2)</p>

<p>Voucher for free shipping show <b>0.00%</b> instead of <b>free shipping</b>.</p>

<p>Best regards</p>

<p>PN</p>

<p> </p>

**Comment by apcloic** [comment]
> point 3 is not corrected. (PS versios 1.7.4.2)
> Voucher for free shipping show 0.00% instead of free shipping.

Hi, any update regarding this issue ?
I'm using PS 1.7.5.0

**Comment by khouloudbelguith** [comment]
Hi @apcloic,

The issue exists also in the PS1.7.5.1, it is added to our bug roadmap.
If you have already fixed it on your end or if you think you can do it, please do send us a pull request!
Thanks!

**Comment by khouloudbelguith** [comment]
Hi,

Here's a PR: https://github.com/PrestaShop/PrestaShop/pull/13645 to fix the issue solved by @idnovate.
Thanks!

**Comment by khouloudbelguith** [comment]
Hi,

Another PR: https://github.com/PrestaShop/PrestaShop/pull/15459 created by @LedCloud.

Thanks!

**Comment by LedCloud** [comment]
I think that there is a mess in the options for the voucher. The voucher can be made at the same time for 1) free delivery, 2) discount value or percent, 3) gift.

Maybe it would be better to create a voucher for only one option of those? It will be simpler to display and describe it to the customer, I think.

**Comment by matthieu-rolland** [comment]
@colinegin @marionf 

I agree with @LedCloud on this one, being able to mix free shipping, reduction and gift in the same voucher is quite confusing in the end. I can still fix this issue, but maybe we could think about improving the feature in the future?

**Comment by colinegin** [comment]
Hello @matthieu-rolland and @LedCloud ,
thanks both for your suggestions ! I quite agree, there's a lot of work to be done with the cart rule creation and management. Unfortunately, this page needs to be migrated first before we can think of improving some features. 

My recommendation is to apply this fix anyway as it fixes an issue on front office, and then we'll think of the expected behaviour and features in BO. Is this ok for you @matthieu-rolland ?

@LedCloud I would be very interested in having your feedbacks on the current cart rule page, to better understand what are the current pain points. Do you think we could arrange a phone call ? 

Thanks !

**Comment by LedCloud** [comment]
@colinegin 

I'm not sure I can be usefull for you. The exist solution fits me (as a merchant) fine. And for I can fix this issue for my store I have no suggestion. At least for the moment.

**Comment by colinegin** [comment]
Alright, no problem :)
